src/utils/integrationSrc.ts

```diff
@@ -0,0 +1,19 @@
+export function addIntegrationSrc<T extends { integration?: string; }>(obj: T | undefined, intgSrc: string | undefined): T | undefined {
+    if (!intgSrc) {
+        return obj;
+    }
+
+    if (!obj) {
+        return {
+            integration: intgSrc
+        } as T;
+    }
+
+    if (obj.integration) {
+        // do not override if integration already exist
+        return;
```

Comment:
This function expects a value to be returned, should this return `obj`? 

Or this could be 
``` typescript
if (!obj.integration) {
  obj.integration = intgSrc;
}
return obj;
```

Reply:
off yes good catch